### PR TITLE
Week 2 rashaan richardson

### PR DESCRIPTION
**Pull request by Rashaan-Richardson**
# Submitting Week 2 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [ ] Coding Challenges
- [ ] Coding Concepts
- [ ] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -15,19 +15,42 @@
...
+    return arr.map(num => num * 3)
+}
+
 
 // --------------------1) Create a function that takes a object as an argument and decides if the number inside it is evenly divisible by three or not.
 
 // Pseudo code:
+// input: object
+// output : string containing a conditional
+// process: Will use a ternary operator to iterate through our object
```
Comment (chrisyhlee) [inline code-challenges.test.js:45]: The ternary operator does not iterate, it is just a short hand way to write if..else statements.

File: code-challenges.test.js [hunk 2]
```diff
@@ -15,19 +15,42 @@
...
 
 // a) Create a test with expect statements for each of the variables provided.
+// describe("even3", () => {
+//     it("Create a function that takes a object as an argument and decides if the number inside it is evenly divisible by three or not.", () => {
+//         const object1 = { number: 15 }
+//         expect(even3(object1)).toEqual([0])
```
Comment (chrisyhlee) [inline code-challenges.test.js:51]: What about the expect statements for the other 2 given variables?

File: code-challenges.test.js [hunk 3]
```diff
@@ -38,9 +61,19 @@ const object3 = { number: -7 }
...
+console.log(even3(object2));
+console.log(even3(object3));
+
 // --------------------2) Create a function that takes in an array of words and returns an array with all the words capitalized.
 
-// Pseudo code:
+// Pseudo code: 
+// input: array that contains strings
+// output : string with the first letter capitilized
+// process: Will use indexing and toUpperCase() to change the string
```
Comment (chrisyhlee) [inline code-challenges.test.js:76]: There is no mention of `.map` or `.replace` and how they work in your pseudo code!

File: code-challenges.test.js [hunk 4]
```diff
@@ -49,12 +82,29 @@ const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
...
+console.log(titleCase(randomNouns1));
+console.log(titleCase(randomNouns2));
 
 // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
 
 // Pseudo code:
 
+// input: strings
+// output : the index of the first vowel in the string
+// process: Will use regex and .search. Since that returns the index of the match
```
Comment (chrisyhlee) [inline code-challenges.test.js:106]: This pseudo code is better than the last two challenges! I would elaborate some more though. Along with what you're using, how does it work?

File: code-challenges.test.js [hunk 5]
```diff
@@ -64,4 +114,18 @@ const vowelTester2 = "academy"
...
+// describe("vowelLog", () => {
+//     it("CCreate a function that takes in a string and logs the index of the first vowel", () => {
+        // const vowelTester1 = "learn"
+    //     expect(vowelTester1(object1)).toEqual(object1[1])
+//     })
+// })
 // b) Create the function that makes the test pass.
+
+const vowelLog = (object) => {
+    return object.search(/[aeiou]/g)
```
Comment (chrisyhlee) [inline code-challenges.test.js:126]: The 'g' flag in the regex is not necessary; the 'i' flag would be what you want for edge cases.

File: coding-concepts.js [hunk 6]
```diff
@@ -21,23 +24,32 @@ const greeter = (name) => {
 // console.log(greeter("LEARN Student"))
 
 // a) Your answer:
+// Hello Learn Student
+
 // b) Verify and explain:
+// The greeter function takes in parameter and returns it via string interperlation.
```
Comment (chrisyhlee) [inline coding-concepts.js:30]: It logs `undefined` because there is no return in the function

**Review by chrisyhlee — APPROVED** [review]
Rashaan, good work on assessment 2. This is currently showing Novice level of comprehension. With a little improvement on your pseudo code and basic concepts, you will be at Advanced Beginner. I would say the main focus for you is making sure your pseudo is fully explaining every part of your code. What tools are you using, and how is it turning the input to the output? Keep up the good work!
